pkg/compiler/types.go

```diff
@@ -830,7 +830,7 @@ var typeFmt = &typeDesc{
 	Check: func(comp *compiler, t *ast.Type, args []*ast.Type, base prog.IntTypeCommon) {
 		desc, _, _ := comp.getArgsBase(args[1], true)
 		switch desc {
-		case typeResource, typeInt, typeLen, typeFlags, typeProc:
+		case typeResource, typeInt, typeLen, typeFlags, typeProc, typeConst:
```

Comment:
I am not sure about this. 
`fmt[hex, const[0x1]]` looks like a strange way to create a fixed string, which is easier with `stringnoz` type. This is especially suspicious given the current limitations of fmt type (fixed-size result).
In lots of cases when people try to do something like this, they are confused and don't really realize what they are doing.
I would prefer to wait for some real world use case for this first, and remove "const" from documentation instead.

Reply:
Personally, I do not need this change, it was only there because of inconsistency between docs and code in the first place. I do not have a strong opinion on this matter. However, the reason I think it is better to have fmt+const is artificial, but it seems like a much cleaner way if you need constant in two different formats, i.e:

test.txt.const:
`ID = 66`

test.txt
```
data_1 {
	name_with_dec_id	"somename"
	field_0			fmt[dec, const[ID]]
	oz			        const[0x00]
}

data_2 {
	name	"somename"
	id	        fmt[hex, const[ID]]
        oz	        const[0x00]
}
```

But I understand the reasoning and added suggested change instead of mine. Considering the limitations, do you think it would be a good enhancement to have a format specifier for 'fmt'? It was somewhat beneficial for me to have that recently, but in my case it was much easier to patch the compiler itself in order to generate string with different size.

Reply:
It's hard to support arbitrary format strings. The main problem is not different size, but supporting dynamic size. Current implementation doesn't support it at all.